client/components/DataManagement/filterSortHooks.js

```diff
@@ -73,34 +73,41 @@ export const useDerivedOverallPhaseByTestPlanId = (
                     phase,
                     timeKey
                 );
+                // Gather all of the phases for a test plan
+                // The first element in the array is the overall phase
+                // while the subsequent elements in the array are TestPlanVersions
+                // that are in other phases
                 if (derivedPhase) {
-                    derivedOverallPhaseByTestPlanId[testPlan.id] = derivedPhase;
-                    break;
+                    let arr = activeTestPlanVersionsByPhase[testPlan.id];
+                    if (!arr) activeTestPlanVersionsByPhase[testPlan.id] = [];
+                    activeTestPlanVersionsByPhase[testPlan.id].push(
+                        derivedPhase
+                    );
```

Comment:
The way that the reference is being accessed/modified is a little slippery to me here.

I wonder if something without the temporary var could also work well.

```
  // Ensure the array exists or initialize it directly before pushing the new phase
  activeTestPlanVersionsByPhase[testPlan.id] = activeTestPlanVersionsByPhase[testPlan.id] ?? [];
  activeTestPlanVersionsByPhase[testPlan.id].push(derivedPhase);

```